plugins/fastlane/src/index.ts

```diff
@@ -0,0 +1,154 @@
+import {
+  Auto,
+  IPlugin,
+  execPromise,
+  validatePluginConfiguration,
+  ILogger,
+  getPrNumberFromEnv,
+  DEFAULT_PRERELEASE_BRANCHES,
+  getCurrentBranch,
+  determineNextVersion,
+} from "@auto-it/core";
+
+import { inc, ReleaseType } from "semver";
+
+import * as t from "io-ts";
+
+import { getpListContents, writepListContents } from "./utilities";
+
+const logPrefix = "[Fastlane-Plugin]";
+
+/**
+ * Wrapper to add logPrefix to messages
+ *
+ * @param msg - The message to add a prefix to
+ * @returns Log message with prefix prepended
+ */
+const logMessage = (msg: string): string => `${logPrefix} ${msg}`;
+
+const required = t.interface({
+  /** Relative path to podspec file */
+  pListPath: t.union([t.string, t.array(t.string)]),
+});
+
+const optional = t.partial({
+  /** The command to use for `fastlane` if it needs to be separate like `fastlane action increment_version_numer` */
+  fastlaneCommand: t.string,
+  /** xcode project path */
+  xcodeproj: t.string,
```

Comment:
i wouldnt put this as optional, since it would be required for `fastlane` to work purely through `fastlane action`